consistent-sampling/src/main/java/io/opentelemetry/contrib/samplers/ConsistentSampler.java

```diff
@@ -47,8 +48,25 @@ public static final ConsistentSampler alwaysOff() {
    * @param samplingProbability the sampling probability
    * @return a sampler
    */
-  public static final ConsistentSampler probabilityBased(double samplingProbability) {
-    return new ConsistentProbabilityBasedSampler(samplingProbability);
+  public static ConsistentSampler probabilityBased(double samplingProbability) {
+    RandomGenerator randomGenerator = RandomGenerator.getDefault();
+    return new ConsistentProbabilityBasedSampler(
+        samplingProbability,
+        s -> randomGenerator.numberOfLeadingZerosOfRandomLong(),
+        randomGenerator);
+  }
+
+  /**
+   * Returns a {@link ConsistentSampler} that samples each span with a fixed probability.
+   *
+   * @param samplingProbability the sampling probability
+   * @param rValueGenerator the function to use for generating the r-value
```

Comment:
Probably needs some more documentation. The `rValueGenerator` must map a trace ID to a value in range [0,62] with probabilities as specified in https://opentelemetry.io/docs/reference/specification/trace/tracestate-probability-sampling/#appendix. I would prefer a new functional interface instead of `ToIntFunction<String>` where all this can be documented.

Reply:
I agree. A new interface could be also used to define the default behavior "s -> RandomGenerator.getDefault().numberOfLeadingZerosOfRandomLong()" and avoid repeating this code phrase in multiple places.

Reply:
Great ideas, implemented. I put the default behavior in a separate class `RValueGenerators` to keep it package-private.